src/test/testing_assert.cpp

```diff
@@ -42,18 +56,65 @@ Matrix _table_to_matrix(const std::shared_ptr<const opossum::Table>& t) {
   return matrix;
 }
 
-std::string _matrix_to_string(const std::string& title, const Matrix& m) {
+std::string _matrix_to_string(const Matrix& matrix, const std::vector<std::pair<uint64_t, uint16_t>>& highlight_cells,
+                              const std::string& highlight_color) {
   std::stringstream stream;
-  stream << "-------" << title << "-------" << std::endl;
-  for (unsigned row = 0; row < m.size(); row++) {
-    for (opossum::ColumnID col{0}; col < m[row].size(); col++) {
-      stream << std::setw(8) << m[row][col] << " ";
+  bool highlight;
+  bool previous_row_highlighted = false;
+
+  std::string highlight_color_bg = ANSI_COLOR_BG_RED;
+  if (highlight_color == ANSI_COLOR_GREEN) {
```

Comment:
I am not a fan of using color codes as some type of logical condition. If this is necessary in this case, please comment.

Reply:
No you're right, not necessary, I changed it.